spigot/components/src/main/kotlin/com/mineinabyss/geary/minecraft/components/DisplayBossBar.kt

```diff
@@ -17,6 +21,8 @@ import org.bukkit.boss.BarColor
 public class DisplayBossBar(
     public val color: BarColor,
     public val style: BarStyle,
-    public val range: Double
+    public val range: Double,
+    public var playersInRange: Set<Player>
```

Comment:
This should be marked as `@Transient` as well, and probably makes more sense being inside the class and not constructor. Also make it a `val` but use `MutableSet` so you can .add and remove from it.

One last thing, players should probably be stored as UUIDs because sometimes the `Player` reference gets broken over time. To do this, in your system you could do:
```kotlin
location.getNearbyPlayers(bossbar.range).map { it.uniqueId }
```